### PR TITLE
Migrate in_monitor_agent to v0.14 API

### PR DESCRIPTION
**Pull request by cosmo0920**
I've migrated in_monitor_agent plugin to v0.14 API.

### Remaining tasks

- [x] Write more test cases for in_monitor_agent

### Not for this PR tasks

* support more plugin classes in in_monitor_agent (e.g. parser and storage plugins support)

### CODE REVIEW

**Comment by tagomoris** [comment]
More test cases are very good for this change.
But I think it's not time to add new features... it should be done in another following pull-requests.

**Comment by cosmo0920** [comment]
`MultiOutput` class should belong to `output` plugin category?
Or, leave it as-is for now?

**Comment by tagomoris** [comment]
Leave it as-is. It's far different from normal output plugins.

**Comment by cosmo0920** [comment]
I've tried to add test cases for in_monitor_agent.

File: lib/fluent/plugin/in_monitor_agent.rb [hunk 1]
```diff
@@ -341,10 +292,10 @@ def self.collect_children(pe, array=[])
     # try to match the tag and get the info from the matched output plugin
     # TODO: Support output in label
     def plugin_info_by_tag(tag, opts={})
-      matches = Engine.root_agent.event_router.instance_variable_get(:@match_rules)
+      matches = Fluent::Engine.root_agent.event_router.instance_variable_get(:@match_rules)
       matches.each { |rule|
         if rule.match?(tag)
-          if rule.collector.is_a?(Output)
+          if rule.collector.is_a?(Fluent::Output)
```
Comment (tagomoris) [inline lib/fluent/plugin/in_monitor_agent.rb]: `Fluent::Plugin::Output` looks correct instead of `Fluent::Output`(this means v0.12 plugin instances only).

Comment (cosmo0920) [inline lib/fluent/plugin/in_monitor_agent.rb]: Oh, I've missed it. 😰 
I'll fix it.

**Comment by cosmo0920** [comment]
I've tried to reduce too strict assertions and add tests.
Please review this PR once again.

File: lib/fluent/plugin/in_monitor_agent.rb [hunk 2]
```diff
@@ -330,7 +282,7 @@ def all_plugins
     # from the plugin `pe` recursively
     def self.collect_children(pe, array=[])
       array << pe
-      if pe.is_a?(MultiOutput) && pe.respond_to?(:outputs)
```
Comment (tagomoris) [inline lib/fluent/plugin/in_monitor_agent.rb]: It is required to collect data for old MultiOutput plugins.
Old (v0.12) MultiOutput plugin are not children of `Fluent::Plugin::MultiOutput`, but `Fluent::Compat::MultiOutput` (and it is child of `Fluent::Plugin::BareOutput`).
fluent-plugin-forest and other plugins are affected with it.

Comment (cosmo0920) [inline lib/fluent/plugin/in_monitor_agent.rb]: Oh, I see....
I've added additional commit to resolve your commented problem: https://github.com/fluent/fluentd/pull/1151/commits/03cf62e8d1b15ca90a52b979b2d08ec4d06dd6aa

Comment (tagomoris) [inline lib/fluent/plugin/in_monitor_agent.rb]: You should do `pe.is_a?(Fluent::Plugin::MultiOutput) || pe.is_a?(Fluent::MultiOutput)` (both respond to `#otuputs`).

File: lib/fluent/plugin/in_monitor_agent.rb [hunk 3]
```diff
@@ -248,50 +220,29 @@ def start
       @srv.mount('/api/plugins.json', JSONMonitorServlet, self)
       @srv.mount('/api/config', LTSVConfigMonitorServlet, self)
       @srv.mount('/api/config.json', JSONConfigMonitorServlet, self)
-      @thread = Thread.new {
+      thread_create :servlet do
```
Comment (tagomoris) [inline lib/fluent/plugin/in_monitor_agent.rb]: Thread name is process global.
Use thread name with plugin name prefix.

File: lib/fluent/plugin/in_monitor_agent.rb (hunk 2 above)
Comment (tagomoris) [inline lib/fluent/plugin/in_monitor_agent.rb]: @cosmo0920 ping?

Comment (cosmo0920) [inline lib/fluent/plugin/in_monitor_agent.rb]: pong. I've pushed a commit to fix this.

File: lib/fluent/plugin/in_monitor_agent.rb [hunk 4]
```diff
@@ -341,10 +293,10 @@ def self.collect_children(pe, array=[])
     # try to match the tag and get the info from the matched output plugin
     # TODO: Support output in label
     def plugin_info_by_tag(tag, opts={})
-      matches = Engine.root_agent.event_router.instance_variable_get(:@match_rules)
+      matches = Fluent::Engine.root_agent.event_router.instance_variable_get(:@match_rules)
       matches.each { |rule|
         if rule.match?(tag)
-          if rule.collector.is_a?(Output)
+          if rule.collector.is_a?(Fluent::Plugin::Output) || rule.collector.is_a?(Fluent::Output)
```
Comment (cosmo0920) [inline lib/fluent/plugin/in_monitor_agent.rb]: I think that we should both `Fluent::Plugin::Output` and `Fluent::Output` as output plugins, right?

Comment (tagomoris) [inline lib/fluent/plugin/in_monitor_agent.rb]: Right.

**Comment by tagomoris** [comment]
LGTM.

**Comment by cosmo0920** [comment]
Thanks! :wink:

**Pull request merged**